cmake/multi_image.cmake

```diff
@@ -267,19 +170,146 @@ function(add_child_image_from_source)
     # above, we only re-run the regex to extract the part after
     # '_'. We run the regex twice because it is believed that
     # list(FILTER is faster than doing a string(REGEX on each item.
-    string(REGEX MATCH "^${ACI_NAME}_(.+)" unused_out_var ${var_name})
+    string(REGEX MATCH "^${GET_ARGS_CHILD_IMAGE}_(.+)" unused_out_var ${var_name})
 
     # When we try to pass a list on to the child image, like
     # -DCONF_FILE=a.conf;b.conf, we will get into trouble because ; is
     # a special character, so we escape it (mucho) to get the expected
     # behaviour.
     string(REPLACE \; \\\\\; val "${${var_name}}")
 
-    list(APPEND image_cmake_args
+    list(APPEND child_image_cmake_args
       -D${CMAKE_MATCH_1}=${val}
       )
   endforeach()
 
+  unset(${GET_ARGS_CONFIG_OUT} PARENT_SCOPE)
+  set(${GET_ARGS_CONFIG_OUT} "${child_image_cmake_args}" PARENT_SCOPE)
```

Comment:
unset(a)
set(a foo)

is the same as

set(a foo)

no?

Reply:
I would think so, I'll remove it.